lib/mistral_ai_summary_example/settings_dialogs.dart

```diff
@@ -0,0 +1,262 @@
+      title: 'Random Seed',
+      description: 'Write a number if you want to set a random seed',
+      onSettingChanged: () {
+        final randomSeed = _randomSeedController.text.isEmpty
+            ? null
+            : int.tryParse(_randomSeedController.text);
+        context.read<SummarySettingsModel>().setRandomSeed(randomSeed);
+        Navigator.of(context).pop();
+      },
+      child: Column(
+        children: [
+          const SizedBox(height: 24),
+          TextField(
+            controller: _randomSeedController,
+            onChanged: (_) => setState(() {}),
+            decoration: InputDecoration(
+              border: const OutlineInputBorder(),
+              suffixIcon: InkWell(
+                borderRadius: BorderRadius.circular(20),
+                onTap: () => setState(_randomSeedController.clear),
+                child: const Icon(Symbols.cancel),
+              ),
+              hintText: 'e.g. 123',
+            ),
+          ),
+          const SizedBox(height: 16),
+          Row(
+            mainAxisAlignment: MainAxisAlignment.spaceBetween,
+            children: [
+              const Text(
+                'Unset',
+              ),
+              Switch(
+                value: _randomSeedController.text.isEmpty,
+                onChanged: null,
+                activeTrackColor: Theme.of(context).colorScheme.primary,
+              ),
+            ],
+          ),
+          const SizedBox(height: 24),
+        ],
+      ),
+    );
+  }
+}
+
+class CommonSettingsDialog extends StatelessWidget {
+  const CommonSettingsDialog({
+    required this.title,
+    required this.description,
+    required this.child,
+    required this.onSettingChanged,
+    super.key,
+  });
+
+  final String title;
+  final String description;
+  final Widget child;
+  final VoidCallback onSettingChanged;
+
+  @override
+  Widget build(BuildContext context) {
+    return Dialog(
+      child: SizedBox(
+        width: 312,
+        child: Padding(
+          padding: const EdgeInsets.all(24),
+          child: Column(
+            crossAxisAlignment: CrossAxisAlignment.start,
+            mainAxisSize: MainAxisSize.min,
+            children: [
+              Text(
+                title,
+                style: Theme.of(context).textTheme.bodyLarge,
+              ),
+              const SizedBox(height: 8),
+              Text(
+                description,
+                style: Theme.of(context).textTheme.bodyMedium,
+              ),
+              child,
+              Align(
+                alignment: Alignment.centerRight,
+                child: Wrap(
+                  children: [
+                    TextButton(
+                      onPressed: () => Navigator.of(context).pop(),
+                      child: const Text('Cancel'),
+                    ),
+                    TextButton(
+                      onPressed: onSettingChanged,
+                      child: const Text('OK'),
```

Comment:
I would change the name here to something like `Update` or `Change`. `OK` seems not the best choice for me.

Reply:
`OK` may be misleading. What do you think about `apply`? I think it is popular in the context of settings